### DISCUSSION TITLE
Is it possible to log external request hostnames?

### DISCUSSION THREAD

**Discussion by dhf22** [istio/istio#54165] [Q&A]
Currently running istio v1.24.1 in sidecar mode
I am trying to log all external requests from the cluster using the following envoy config. When I examine the logs I am able to see the IP address of the external request from UPSTREAM_HOST however there is no visibility of the actual hostname. I've seen posts indicating that since there is no TLS termination on the sidecar, istio is unable to get details of the request, wanted to confirm that this is still the case even with envoy now supporting UPSTREAM_HOST_NAME. Thanks in advance
```          
  values:
    meshConfig:
      accessLogEncoding: "JSON"
      accessLogFile: "/dev/stdout"
      extensionProviders:
            - name: envoy
              envoyFileAccessLog:
                path: /dev/stdout
                logFormat:
                  labels:
                    upstream_host: "%UPSTREAM_HOST%"
                    upstream_host_name: "%UPSTREAM_HOST_NAME%"

**Comment by howardjohn** [comment]
https://canistiobreaktls.com/ -- no, if its TLS encrypted it cannot see the hostname. In some cases you can see `REQUESTED_SERVER_NAME` but often not because https://github.com/envoyproxy/envoy/issues/34396 is not implemented

**Reply by dhf22** [reply]
appreciate the reply
is it possible to retrieve hostname of an external requests from SNI with TLS inspector? I attempted to do this but REQUESTED_SERVER_NAME was still null, not sure if my implementation was incorrect or just not possible to do consistently?

**Reply by howardjohn** [reply]
Yes but its not done in Istio because its not safe because https://github.com/envoyproxy/envoy/issues/34396 is not implemented
